enterprise_access/apps/api/serializers.py

```diff
@@ -207,67 +192,136 @@ def validate(self, attrs):
 
-    class Meta:
-        model = SubsidyAccessPolicy
-        exclude = ('created', 'modified')
+class SubsidyAccessPolicyListRequestSerializer(ValidateContentKeyMixin, serializers.Serializer):
+    """
+    Request Serializer to validate policy list endpoint query params.
 
-    def get_policy_redemption_url(self, obj):
-        location = reverse('api:v1:policy-redeem', kwargs={'policy_uuid': obj.uuid})
-        return get_current_request().build_absolute_uri(location)
+    Request serializer for views:
+    * SubsidyAccessPolicyRedeemViewset.list
+    """
+    lms_user_id = serializers.IntegerField(required=True)
+    content_key = serializers.CharField(required=True)
+    enterprise_customer_uuid = serializers.UUIDField(required=True)
 
 
-class SubsidyAccessPolicyCreditAvailableSerializer(SubsidyAccessPolicyRedeemableSerializer):
-    """
-    Serializer to transform response for policy redeem GET endpoint.
+class SubsidyAccessPolicyRedemptionRequestSerializer(ValidateContentKeyMixin, serializers.Serializer):
     """
-    remaining_balance_per_user = serializers.SerializerMethodField()
-    remaining_balance = serializers.SerializerMethodField()
-
-    def get_remaining_balance_per_user(self, obj):
-        learner_id = self.context.get('learner_id')
-        return obj.remaining_balance_per_user(learner_id=learner_id)
+    Request Serializer to validate policy redemption endpoint query params.
 
-    def get_remaining_balance(self, obj):
-        return obj.remaining_balance()
+    Request serializer for views:
+    * SubsidyAccessPolicyRedeemViewset.redemption
+    """
+    lms_user_id = serializers.IntegerField(required=True)
+    content_key = serializers.CharField(required=True)
+    enterprise_customer_uuid = serializers.UUIDField(required=True)
 
 
-class SubsidyAccessPolicyCreditAvailableListSerializer(serializers.Serializer):  # pylint: disable=abstract-method
+class SubsidyAccessPolicyCreditsAvailableRequestSerializer(serializers.Serializer):  # pylint: disable=abstract-method
     """
     Serializer to validate policy request GET query params.
+
+    Request serializer for views:
+    * SubsidyAccessPolicyRedeemViewset.credits_available
     """
     enterprise_customer_uuid = serializers.UUIDField(required=True)
-    lms_user_id = serializers.CharField(required=True)
+    lms_user_id = serializers.IntegerField(required=True)
 
 
 class SubsidyAccessPolicyCanRedeemRequestSerializer(serializers.Serializer):  # pylint: disable=abstract-method
     """
     Serializer to validate SubsidyAccessPolicyRedeemViewset.can_redeem GET request parameters.
+
+    Request serializer for view:
+    * SubsidyAccessPolicyRedeemViewset.can_redeem
     """
     enterprise_customer_uuid = serializers.UUIDField(required=True)
     content_key = serializers.ListField(child=serializers.CharField(required=True), allow_empty=False)
 
-    def validate_content_key(self, value):
+    def validate_content_key(self, content_keys):
```

Comment:
[clarification] Is there an opportunity to re-use `ValidateContentKeyMixin` for this serializer?

Reply:
oh, i just forgot